cpmpy/expressions/python_builtins.py

```diff
@@ -125,3 +127,19 @@ def sum(*iterable, **kwargs):
 
     assert len(kwargs)==0, "sum over decision variables does not support keyword arguments"
     return Operator("sum", iterable)
+
+
+def abs(element):
+    """
+        abs() overwrites the python built-in to support decision variables.
+
+        if the element given is not a CPMpy expression, the built-in is called
+        else an Absolute functional global constraint is constructed.
+    """
+    if is_any_list(element):
+        return cpm_array([abs(elem) for elem in element])
+
+    if isinstance(element, Expression):
```

Comment:
is this missing a 'not'??? (or returning the wrong thing)

Reply:
ah this actually works (for the same reason `from cpmpy import *` works, because our expressions override the builtin). but of course the not should be there, so we use the builtin for non-expressions.

This makes me think that we can maybe remove the abs code from the expressions themselves, since everything will go through this function now

Reply:
indeed the not was missing